include/pybind11/async.h

```diff
@@ -0,0 +1,208 @@
+#pragma once
+
+#include <memory>
+#include <future>
+#include <chrono>
+
+#include "pybind11/pybind11.h"
+
+
+namespace py = pybind11;
+
+
+PYBIND11_NAMESPACE_BEGIN(PYBIND11_NAMESPACE)
+PYBIND11_NAMESPACE_BEGIN(async)
+
+
+class StopIteration : public py::stop_iteration {
+    public:
+        StopIteration(py::object result) : stop_iteration("--"), result(result) {};
+        //using py::stop_iteration::stop_iteration;
+        /// Set the error using the Python C API
+        void set_result(py::object result) {
+            this->result = result;
```

Comment:
```suggestion
            this->result = std::move(result);
```
or make it take a const ref instead.

Reply:
thanks for the hint! it wasn't used so i removed it altogether...